### DISCUSSION TITLE
Proper way of configuring keywords with Node.js SDK

### DISCUSSION THREAD

**Discussion by lminiero** [deepgram/community#107] [General help]
Hi,

I've been tinkering with Deepgram for a while, and so far it's been working surprisingly well. There's a feature I haven't been able to activate or use properly, though, which are keywords. I'm using the Node.js `@deepgram/sdk`, feeding it Opus chunks of audio I'm proxying myself from a WebRTC server.

My usage of the SDK is fairly basic, since I configure it like that (only keeping the relevant bits in):

```
const { Deepgram } = require('@deepgram/sdk');
let deepgram = new Deepgram(apiKey);
let options = {
	language: 'en-US',
	channels: 1,
	sample_rate: 16000,
	encoding: 'opus',
	punctuate: true,
	meeting: true,
	diarize: true,
	interim_results: true
};
let session = deepgram.transcription.live(options);
```

I'm omitting the listeners for the sake of brevity, also because everything does work.

Now, looking at the documentation, there's ambiguous definitions for how keywords should be provided when added to the `options` object above: the official documentation only covers how to provide them via query string, while the page on npm seems to suggest an array of strings should be used instead; a different source suggested using an array of `{ word: <string>, boost: <number> }` objects instead. I tried using both but to no avail, e.g.,

```
let options = {
	language: 'en-US',
	channels: 1,
	sample_rate: 16000,
	encoding: 'opus',
	punctuate: true,
	meeting: true,
	diarize: true,
	interim_results: true,
	keywords: [ 'pippo:2.0', topolino:2.0', 'paperino:2.0' ]
}
```

or

```
...
	keywords: [
		{ word: 'pippo', boost: 2.0 },
		{ word: 'topolino', boost: 2.0 },
		{ word: 'paperino', boost: 2.0 }
	]
...
```

but neither seemed to have any effect (I used different words too, the ones above are just an example) when I tried transcribing live audio that included those words. I also tried adding a `tier: 'enhanced'` property to the options, as some sources seemed to suggest keywords needed that, or at the very least worked better when it was enabled, but that didn't seem to make any difference.

My main objective is to be able to properly detect some context specific terminology that makes a heavy use of acronyms, which I assume will be a challenge in itself (e.g., detecting something like "SMTP" when people spell the individual letters as you'd usually pronounce it), but at the moment I'm unable to add even custom words to that, which makes me think I'm doing something wrong. 
Taking into account the snippets above, am I configuring this improperly, or are there different requirements for this to actually work? Not sure for instance if keywords only work when using custom models in premium subscriptions, or if they're supposed to work when using the pay-as-you-go plans too. Is there any way to query a currently active Deepgram session and check whether keywords are actually active, after it's been created?

Thanks in advance for any help you'll be able to provide! Please do let me know if there's more info I should add.

**Comment by briancbarrow** [comment]
Hi @lminiero, The correct way to do it is like this:
```
let options = {
	language: 'en-US',
	channels: 1,
	sample_rate: 16000,
	encoding: 'opus',
	punctuate: true,
	meeting: true,
	diarize: true,
	interim_results: true,
	keywords: [ 'pippo:2', topolino:2', 'paperino:2' ]
}
```

So what you were doing was close. I would change it to a whole integer, instead of a decimal number. Try that and let me know if that helps.

**Reply by briancbarrow** [reply]
You can also see that the keywords get passed in properly if you visit the "Usage" page on https://console.deepgram.com/ under "Logs".

**Reply by lminiero** [reply]
Hi @briancbarrow, thanks for the feedback. I checked the logs and it looks like I was indeed passing them more or less correctly in at least some of the attempts, then, but they didn't seem to have any effect nevertheless (maybe because I was using the base model rather than the enhanced tier or a custom model?). I'll make some more tests in the next few days.

**Reply by SandraRodgers** [reply]
Hi @lminiero ,

Actually it has come to my attention that the keywords feature does not work with our enhanced model. So if you want to use keywords, stick with base. 

If keywords aren't working well for you in base, there might be something going on. I know that this feature can be less reliable than other features, such as our search feature. If you still find it totally doesn't work, even with the base model, please let us know.
